Added [:T] syntax explanation to generics tutorial. (#15890)
nim-lang/Nim: doc/tut2.rst

best match of ``add`` is used. The built-in ``add`` procedure for sequences
is not hidden and is used in the ``preorder`` iterator.

There is a special [:T] syntax when using generics with universal function call syntax:

.. code-block:: nim
  proc foo[T](i: T) =

>       discard

Reviewer: ```suggestion
    discard
```

>   var i : int

Reviewer: ```suggestion
  var i: int
```

best match of ``add`` is used. The built-in ``add`` procedure for sequences
is not hidden and is used in the ``preorder`` iterator.

> There is a special [:T] syntax when using generics with universal function call syntax:

Reviewer 2: It's called the "method call syntax", there is nothing "Universal" about special casing the first argument that a function receives.

Author: Noted. Thank you for the comment. I'm very new to Nim and wasn't exactly sure what to write there.

> There is a special [:T] syntax when using generics with universal function call syntax:

Reviewer 2: ```suggestion
There is a special ``[:T]`` syntax when using generics with the method call syntax:
```

.. code-block:: nim
    :test: "nim c $1"
  proc foo[T](i: T) =
    discard

  var i: int

>   #i.foo[int]() #Error: expression 'foo(i)' has no type (or is ambiguous)

Reviewer: ```suggestion
  # i.foo[int]() # Error: expression 'foo(i)' has no type (or is ambiguous)
```

>   i.foo[:int]() #Success

Reviewer: ```suggestion
  i.foo[:int]() # Success
```
